admin/acceptance_tests/layers_wms_test.py

```diff
@@ -289,26 +278,67 @@ def test_duplicate(self, layer_wms_test_data, test_app, dbsession):
 
         ras = layer_wms_test_data['restrictionareas']
         assert set((ras[3].id, ras[0].id)) == set(i.id for i in layer.restrictionareas)
-        self.check_checkboxes(
-            form,
-            'restrictionareas',
-            [{
-                'label': ra.name,
-                'value': str(ra.id),
-                'checked': ra in layer.restrictionareas
-            } for ra in sorted(ras, key=lambda ra: ra.name)])
+        self._check_restrictionsareas_are_checked(form, ras, layer)
+
+        self._check_dimensions_are_duplicated(form, layer)
+
+        self.set_first_field_named(form, 'name', 'clone')
+        resp = form.submit('submit')
+
+        layer = dbsession.query(LayerWMS). \
+            filter(LayerWMS.name == 'clone'). \
+            one()
+        assert str(layer.id) == re.match('http://localhost/layers_wms/(.*)', resp.location).group(1)
+        assert layer.id == layer.dimensions[0].layer_id
+        assert layer.id == layer.metadatas[0].item_id
+        assert layer_wms_test_data['layers'][3].metadatas[0].name == layer.metadatas[0].name
+        assert layer_wms_test_data['layers'][3].metadatas[1].name == layer.metadatas[1].name
+
+    def test_convert(self, layer_wms_test_data, test_app, dbsession):
+        from c2cgeoportal_commons.models.main import LayerWMTS
+        layer = layer_wms_test_data['layers'][3]
+
+        resp = test_app.get("/layers_wmts/from_wms/{}".format(layer.id), status=200)
 
+        form = resp.form
+        assert '' == self.get_first_field_named(form, 'id').value
+        assert layer.name == self.get_first_field_named(form, 'name').value
+        assert str(layer.metadata_url or '') == form['metadata_url'].value
+        assert str(layer.description or '') == self.get_first_field_named(form, 'description').value
+        assert layer.public is True
+        assert layer.public == form['public'].checked
+        assert str(layer.geo_table or '') == form['geo_table'].value
+        assert str(layer.exclude_properties or '') == form['exclude_properties'].value
+        assert str(layer.layer or '') == form['layer'].value
+        assert str(layer.style or '') == form['style'].value
+        interfaces = layer_wms_test_data['interfaces']
+        self._check_interfaces_are_checked(form, interfaces, layer)
+        ras = layer_wms_test_data['restrictionareas']
+        self._check_restrictionsareas_are_checked(form, ras, layer)
+        self._check_dimensions_are_duplicated(form, layer)
+
+        self.set_first_field_named(form, 'name', 'duplicated')
+        resp = form.submit('submit')
+
+        layer = dbsession.query(LayerWMTS). \
+            filter(LayerWMTS.name == 'duplicated'). \
+            one()
+        assert str(layer.id) == re.match('http://localhost/layers_wmts/(.*)', resp.location).group(1)
+
+    def test_convert_image_type_from_ogcserver(self, layer_wms_test_data, test_app):
+        layer = layer_wms_test_data['layers'][2]
```

Comment:
Can you explain why there is a specific test for image_type.

Reply:
renamed test_convert into test_convert_common_fields_copied